include/kickcat/Frame.h

```diff
@@ -49,24 +50,31 @@ namespace kickcat
 
         /// \return true if datagram can be extracted, false otherwise
         bool isDatagramAvailable() const { return is_datagram_available_; }
+        void setIsDatagramAvailable() { is_datagram_available_ = true; }
 
-        // handle bus access
-        void read(std::shared_ptr<AbstractSocket> socket);
-        void write(std::shared_ptr<AbstractSocket> socket);
+        void setSourceMAC(mac const src_mac);
 
         // helper to access raw frame (mostly for unit testing)
         uint8_t* data() { return frame_.data(); }
+        EthercatHeader* header()   { return header_; }
+        EthernetHeader* ethernet() { return ethernet_; }
 
+
+
+        /// For writeThenRead
+        uint8_t* last_datagram_{nullptr};   // Last **written** datagram
```

Comment:
Why do you need direct access to member? wirteThenRead() doesn't seem to need tsuch access

Reply:
I don't need it anymore